planner/core/src/main/java/eu/seaclouds/platform/planner/core/Planner.java

```diff
@@ -105,6 +108,7 @@ private String generatMMOutput(Map<String, HashSet<String>> mmResult, Map<String
         return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(singleRes);
     }
 
+
     public String plan() throws ParsingException, IOException {
```

Comment:
Why this method doesn't have any parameters?
offerings, matchamaker, optimizeClient look like good candidates. This will enormously improve the testability of this method, just to mention a benefit